dcos/cosmospackage.py

```diff
@@ -37,10 +37,14 @@ def enabled(self):
         # authorized for the command specified, not this endpoint
         except DCOSAuthorizationException:
             return True
-        # all other errors are treated as endpoint not available
+        # allow exception through so we can show user actual http exception
+        # except 404, because then the url is fine, just not cosmos enabled
         except Exception as e:
             logger.exception(e)
-            return False
+            if "HTTP 404: Not Found" in str(e):
```

Comment:
Okay. For my own edification, I take it that requests doesn't return a typed exception with the status code when it is 404. @tamarrow can we confirm?